scene/gui/popup_menu.cpp

```diff
@@ -421,20 +415,52 @@ void PopupMenu::_activate_submenu(int p_over, bool p_by_keyboard) {
 		DisplayServer::get_singleton()->window_set_popup_safe_rect(submenu_popup->get_window_id(), safe_area);
 	}
 
-	this_rect.position -= submenu_pum->get_position(); // Make the position of the parent popup relative to submenu popup.
+	this_rect.position -= submenu_popup->get_position(); // Make the position of the parent popup relative to submenu popup.
 	this_rect.size.width -= panel_ofs_start.x + panel_ofs_end.x;
 	this_rect.size.height -= panel_ofs_end.y + (theme_cache.panel_style->get_margin(SIDE_TOP) + theme_cache.panel_style->get_margin(SIDE_BOTTOM)) * win_scale;
 
 	// Autohide area above the submenu item.
-	submenu_pum->clear_autohide_areas();
-	submenu_pum->add_autohide_area(Rect2(this_rect.position.x, this_rect.position.y - theme_cache.panel_style->get_margin(SIDE_TOP) * win_scale,
+	submenu_popup->clear_autohide_areas();
+	submenu_popup->add_autohide_area(Rect2(this_rect.position.x, this_rect.position.y - theme_cache.panel_style->get_margin(SIDE_TOP) * win_scale,
 			this_rect.size.x, scaled_ofs_cache + scroll_offset + theme_cache.panel_style->get_margin(SIDE_TOP) * win_scale - theme_cache.v_separation / 2));
 
 	// If there is an area below the submenu item, add an autohide area there.
 	if (scaled_ofs_cache + scaled_height_cache + scroll_offset <= control->get_size().height * win_scale) {
 		const int from = scaled_ofs_cache + scaled_height_cache + scroll_offset + theme_cache.v_separation / 2;
-		submenu_pum->add_autohide_area(Rect2(this_rect.position.x, this_rect.position.y + from, this_rect.size.x, this_rect.size.y - from));
+		submenu_popup->add_autohide_area(Rect2(this_rect.position.x, this_rect.position.y + from, this_rect.size.x, this_rect.size.y - from));
+	}
+}
+
+// Focus the previous selectable item.
+// Returns index of selected item or -1 if there are no selectable items.
+// Will refocus the current item if there are no other valid selections in the menu.
```

Comment:
If we want such comment then it should mention that the searching wraps around (otherwise the "previous" item being at a greater index than the currently selected one might be confusing).

Reply:
Sounds good, will add this

Reply:
How's the wording here now?:
```c++
// Focus the previous selectable item, wrapping around if it reaches the start of the list.
// Returns index of focused item or -1 if there are no selectable items.
// Will refocus the current item if there are no other valid selections in the menu.
int PopupMenu::_focus_previous_selectable() {
```

Reply:
> How's the wording here now?

Now it's clear (the comment matches the code/behavior).